db/memtable.cc

```diff
@@ -415,7 +494,7 @@ class MemTableIterator : public InternalIterator {
     return GetLengthPrefixedSlice(key_slice.data() + key_slice.size());
   }
 
-  Status status() const override { return Status::OK(); }
+  Status status() const override { return status_; }
```

Comment:
I think the convention is `!status().ok()` implies `!Valid()`. Can we change `Valid()` to take into account `status_`?

Reply:
Yes, good point, updated.